src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -524,7 +524,7 @@ var ReactCompositeComponentMixin = {
 
     // Distinguish between a props update versus a simple state update
     if (prevParentElement === nextParentElement) {
-      nextContext = inst.context;
+      nextContext = this._processContext(nextUnmaskedContext);
```

Comment:
I'm not even sure that change is required.  I made it because you flagged that file, but @ryanflorence's test doesn't seem to require that line.